app/src/Event/EventController.php

```diff
@@ -1149,6 +1150,74 @@ public function editTracks($friendly_name)
         );
     }
 
+    /**
+     * Upload Data from CSV for this event
+     * @todo Validate & Process uploaded cSV
+     * @param string $friendly_name
+     */
+    public function eventImportCsv($eventSlug)
+    {
+        $config = $this->application->config('oauth');
+        $request = $this->application->request();
+
+        /** @var FormFactoryInterface $factory */
+        $factory = $this->application->formFactory;
+        $form    = $factory->create(new EventImportFormType());
+
+        //use EventFormImportType to validate the CSV is valid
+        //We possibly want to do some extensive data checking
+        //BEFORE we blindly throw data from here to the API
+        if ($request->isPost()) {
+            try {
+                if (isset($_FILES['event_import']['error']['csv_file'])
+                    && $_FILES['event_import']['error']['csv_file'] == UPLOAD_ERR_OK) {
+                    $eventApi = $this->getEventApi();
+                    $event = $eventApi->getByFriendlyUrl($eventSlug);
+                    $handle = fopen($_FILES['event_import']['tmp_name']['csv_file'], "r");
+
+                    while (!feof($handle)) {
+                        $talk = fgetcsv($handle);
+                        $date_start = new \DateTime();
+                        $date_start->setTimestamp(
+                            strtotime($talk[3].' '.$talk[4])
+                        );
+                        $date_end = new \DateTime();
+                        $date_end->setTimestamp(
+                            strtotime($date_start->format($date_end::ATOM).' plus '.$talk[5].' minutes')
+                        );
+
+                        $talk_data = [
+                            'talk_title' => $talk[0],
+                            'talk_description' => $talk[1],
+                            'type' => $talk[7],
+                            'track' => $talk[8],
+                            'language' => $talk[6],
+                            'start_date' => $date_start->format('c'),
+                            'speakers' => [$talk[2]],
+
```

Comment:
Shouldn't either the talk duration or the end-time of the talk be part of this data-array? If neither is used here why did we calculate it in the first place?

Reply:
Especially when the duration will be set by default to 60 minutes...

Reply:
Good catch, yes I dropped the duration field during debugging. Added it back.